tests/Boost_Tests/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+project(Boost_Tests)
+set(CMAKE_CXX_STANDARD 11)
+add_executable(${PROJECT_NAME}
+        main.cpp
+        )
+
+# Specific to Clare's Mac - temporary for testing...
+if (DEFINED ENV{USER})
+    if ("$ENV{USER}" STREQUAL "clare")
+        set(BOOST_ROOT /Users/clare/Documents/develop/tools/boost/boost_1_71_0/
+                CACHE FILEPATH "" FORCE)
+    endif ()
+endif ()
+
+# unit_test_framework
+find_package(Boost 1.1.0 COMPONENTS REQUIRED)
+if (Boost_FOUND)
+
+    include_directories(${Boost_INCLUDE_DIRS})
+    target_link_libraries(${PROJECT_NAME} ${Boost_LIBRARIES})
```

Comment:
With the above component, the link line would look like this.

```cmake
target_link_libraries(${PROJECT_NAME} PRIVATE Boost::unit_test_framework)
```

That eliminates the need for the `include_directories` command 😁

Reply:
That looked good so I just tried it out, and couldn't get it to work...

I think what's happening is that the "single-header" header is only the boost source directory, and is not copied to an install directory.

**Try 1**

When I change the cmake code to this:

```cmake
find_package(Boost 1.1.0 COMPONENTS REQUIRED)
if (Boost_FOUND)
    target_link_libraries(${PROJECT_NAME} PRIVATE Boost::unit_test_framework)
endif ()
```

I get this error:

```
CMake Error at tests/Boost_Tests/CMakeLists.txt:3 (add_executable):
  Target "Boost_Tests" links to target "Boost::unit_test_framework" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
```

**Try 2**

When I make `unit_test_framework` required, and keep the proposed use of `Boost::unit_test_framework` - like this:

```cmake
find_package(Boost 1.1.0 COMPONENTS unit_test_framework REQUIRED)
if (Boost_FOUND)
    target_link_libraries(${PROJECT_NAME} PRIVATE Boost::unit_test_framework)
endif ()
```

I then get this:

```
CMake Error at /Users/clare/Library/Application Support/JetBrains/Toolbox/apps/CLion/ch-0/193.6494.38/CLion.app/Contents/bin/cmake/mac/share/cmake-3.15/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find Boost (missing: unit_test_framework) (found suitable version
  "1.71.0", minimum required is "1.1.0")
```

Reply:
Does that mean you did not install Boost and you are setting BOOST_ROOT to the source directory or did you install Boost from source?

Reply:
I'm not sure if this answers the question, but the directory `/Users/clare/Documents/develop/tools/boost/boost_1_71_0` was unpacked from a download of the source of that release.

On Thursday I did run the following commands to create an installation in an adjacent directory 

```bash
./bootstrap.sh
export boost_installation_prefix=/Users/clare/Documents/develop/tools/boost/boost_1_71_0_install
./b2 --with-test --prefix=$boost_installation_prefix install
```

But for unrelated, reasons, there were problems building Approval Tests with it, so we resorted to the head-only build instead.